lib/hw_node.py

```diff
@@ -66,36 +83,111 @@ def get_salt_cmd(sls, node):
 def power_cycle(node):
     cmd = get_ipmi_cycle_cmd(node['bmc_ip'])
     local = LocalNode()
-    local.shell(cmd, trace=True)
+    try:
+        local.shell(cmd, trace=True)
+    except:
+        BMCException(sys.exc_info()[1])
     if local.status == 0:
         print("node restarted")
         # TODO check retrun results from results
     else:
-        raise Exception(
+        raise BMCException(
             "ipmitool call failed with status[{}], stdout [{}], stderr[{}]".
             format(local.status,
                    local.stdout.rstrip(),
                    local.stderr))
 
 
-# current output after boot
-# -----------------------
-# Welcome to openSUSE Leap 15.0 - Kernel 4.12.14-lp150.11-default (ttyS1).
-#
-#
-# localhost login
-def wait_node_is_ready(node, timeout=5, attempts=120):
-    local = LocalNode()
-    local.wait_for_port(host=node['ip'], timeout=timeout, attempts=attempts)
-
-
-def minimal_needed_configuration(node, timeout=60):
+def read_last_meaningful_line(filepath):
+    """Return last non empty and not auxillary conman line
+        filepath: path to file for finding last line
+        Function is using some magic string which are defined
+        in conman.
+    """
+    tail = LocalNode()
+    res = tail.shell('tail -n 100 ' + filepath,
+                    stop=True, quiet=True, die=False)
+    for str in reversed(tail.stdout.rstrip().splitlines()):
+        if not (str is '' or
+            str.startswith('<ConMan> Console') or
+            re.search("\d\d\d\d-\d\d-\d\d\s+\d\d:\d\d:\d\d\s*$" , str)):
+            return str
+    return 'no_meaningful_line_found'
```

Comment:
Why not just `return None` here?

Reply:
str values instead of  None was used for debug output simplification and separattion between None of start and None from reading data

Reply:
I do not follow, the current implementation of the function never returns None, as fas as I can see.

Reply:
In the function - yes, there is only one None. But external function usage it will be nice to have 2 non-empty values